### ISSUE TITLE
NPE in UDPConnector.stop()

### ISSUE THREAD

**Issue by sbernard31** [eclipse/californium#345]
I faced an NPE :
```
java.lang.NullPointerException
	at org.eclipse.californium.elements.UDPConnector.stop(UDPConnector.java:206)
	at org.eclipse.californium.core.network.CoapEndpoint.stop(CoapEndpoint.java:455)
	at org.eclipse.californium.core.network.CoapEndpoint.start(CoapEndpoint.java:426)
```
in :
```java
String address = socket.getLocalSocketAddress().toString();
if (socket != null) {
    socket.close();
    socket = null;
}
LOGGER.log(Level.CONFIG, "UDPConnector on [{0}] has stopped.", address);
```

**Comment by boaks** [comment]
@sbernard31 
So is this issue closed with PR #347?
(I picked the fix to 1.0.0 and master as well)

**Comment by sbernard31** [comment]
Yep !
